common/djangoapps/student/views.py

```diff
@@ -1677,8 +1677,14 @@ def create_account_with_params(request, params):
     # when the account is created via the browser and redirect URLs.
     should_link_with_social_auth = third_party_auth.is_enabled() and 'provider' in params
 
-    if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
-        params["password"] = pipeline.make_random_password()
+    # if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
+    #     params["password"] = pipeline.make_random_password()
+
+    if params.get('third_party'):
+        if pipeline.running(request):
```

Comment:
We can remove this check as well as validation error after getting confirmed that `pipeline.running(request)` is only used to identify that current request for account creation is made using third party auth.